### DISCUSSION TITLE
Reset odometry in Spot

### DISCUSSION THREAD

**Discussion by jefflsilveira** [bdaiinstitute/spot_ros2#453] [Q&A] [answered]
Hello everybody,

Does the robot have any functionality to reset the robot's odometry?

I mostly use the robot with my own ROS2 workspace running with the spot ros2 driver that I am developing for motion planning and navigation.

It would be very useful to reset the odometry as if I just turned the robot on. This way, before running small experiments I could reset the odometry and use it as a reference frame to set positions for my own controllers, instead of running SLAM or using a MOCAP system.

**Comment by khughes-bdai** [comment, marked as answer]
Hello,

To the best of my knowledge this isn't supported through the Boston Dynamics SDK -- I couldn't find anything about this in their API ([here](https://dev.bostondynamics.com/protos/bosdyn/api/proto_reference) or [here](https://github.com/boston-dynamics/spot-sdk)). 

A workaround could be to publish a second odometry frame that starts at the location of the default `/odom` frame, and do your motion planning with respect to that new frame. When you want to reset the odometry, you could update the location of that frame to be at the `body` frame of the robot.
